### DISCUSSION TITLE
error occurred: fork/exec

### DISCUSSION THREAD

**Discussion by LilTrublMakr** [adnanh/webhook#638] [Q&A]
Hey all. I am having trouble getting webhooks to execute. The file does exist and has execute permissions but when sending the webhook, I receive the following error: `error occurred: fork/exec /home/webhook/test.py: no such file or directory`.

```
root@network:~# cd /home/webhook/
root@network:/home/webhook# ls -al
total 40
drwxr-xr-x 4 root root 4096 May  8 23:15 .
drwxr-xr-x 4 root root 4096 May  8 21:29 ..
-rw-r--r-- 1 root root  127 May  8 21:28 .env
drwxr-xr-x 8 root root 4096 May  8 23:15 .git
-rw-r--r-- 1 root root 3094 May  8 21:28 .gitignore
-rw-r--r-- 1 root root   19 May  8 21:28 README.md
drwxr-xr-x 2 root root 4096 May  8 21:28 cisco
-rwxr-xr-x 1 root root  288 May  8 23:15 test.py
-rw-r--r-- 1 root root  393 May  8 23:06 test.sh
-rw-r--r-- 1 root root  697 May  8 22:43 webhook.json
root@network:/home/webhook#
```

This is the webhook json file:
```json
[
  {
    "id": "test",
    "execute-command": "/home/webhook/test.py",
    "command-working-directory": "/home/webhook",
    "include-command-output-in-response": true,
    "include-command-output-in-response-on-error": true,
    "pass-arguments-to-command": [
      {
        "source": "entire-payload"
      }
    ]
  }
]
```

Python file:
```json
#!/usr/bin/python3

import json
import os
import requests
import sys
from netmiko import ConnectHandler
from dotenv import load_dotenv, dotenv_values

load_dotenv()
config = dotenv_values()

# Parse the JSON
json = json.loads(sys.argv[1])
print (json)
```

As far as I can tell, everything should be correct. Is there something that I am missing?

Thanks

**Comment by adnanh** [comment]
Quick question: Is this happening inside of a Docker container?

**Reply by LilTrublMakr** [reply]
No. I have tried both the `apt install webhook` and building from go versions (I believe that they are both the same version). Both of them are outputting the same error.

**Reply by LilTrublMakr** [reply]
It seems to run a shell script just fine. With my limited testing, this seems to be limited to trying to run a python script directly from the webhook json file.
